front/lib/models/assistant/actions/retrieval.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  DataTypes,
+  ForeignKey,
+  InferAttributes,
+  InferCreationAttributes,
+  Model,
+} from "sequelize";
+
+import { front_sequelize } from "@app/lib/databases";
+import { AssistantAgentConfiguration } from "@app/lib/models/assistant/agent";
+import { DataSource } from "@app/lib/models/data_source";
+import { TimeframeUnit } from "@app/types/assistant/actions/retrieval";
+
+/**
+ * Action Retrieval configuration
+ */
+export class AssistantAgentRetrievalConfiguration extends Model<
+  InferAttributes<AssistantAgentRetrievalConfiguration>,
+  InferCreationAttributes<AssistantAgentRetrievalConfiguration>
+> {
+  declare id: number;
+
+  declare query: "auto" | "none" | "templated";
+  declare queryTemplate: string | null;
+  declare relativeTimeFrame: "auto" | "none" | "custom";
+  declare relativeTimeFrameDuration: number | null;
+  declare relativeTimeFrameUnit: TimeframeUnit | null;
+  declare topK: number;
+
+  declare agentId: ForeignKey<AssistantAgentConfiguration["id"]>;
+}
+AssistantAgentRetrievalConfiguration.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    query: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      defaultValue: "auto",
+    },
+    queryTemplate: {
+      type: DataTypes.TEXT,
+      allowNull: true,
+    },
+    relativeTimeFrame: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      defaultValue: "auto",
+    },
+    relativeTimeFrameDuration: {
+      type: DataTypes.INTEGER,
+      allowNull: true,
+    },
+    relativeTimeFrameUnit: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+    topK: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      defaultValue: 5,
+    },
+  },
+  {
+    modelName: "assistant_agent_retrieval_configuration",
+    sequelize: front_sequelize,
+    hooks: {
+      beforeValidate: (retrieval: AssistantAgentRetrievalConfiguration) => {
+        // Validation for templated Query
+        if (retrieval.query == "templated") {
+          if (retrieval.queryTemplate === null) {
```

Comment:
Turns out the empty fields are not set to `null` but to `undefined`, so this will never be true. Better to use `!retrieval.queryTemplate`